ql/src/java/org/apache/hadoop/hive/ql/stats/ColStatsProcessor.java

```diff
@@ -218,10 +220,13 @@ public int persistColumnStats(Hive db, Table tbl) throws HiveException, MetaExce
       }
 
       start = System. currentTimeMillis();
+      boolean success = false;
       if (tbl != null && tbl.isNonNative() && tbl.getStorageHandler().canSetColStatistics(tbl)) {
-        tbl.getStorageHandler().setColStatistics(tbl, colStats);
+        success = tbl.getStorageHandler().setColStatistics(tbl, colStats);
       }
-      db.setPartitionColumnStatistics(request);
+      success |= db.setPartitionColumnStatistics(request);
```

Comment:
should we still execute this for iceberg tables?

Reply:
👍

Reply:
can we please add a comment linking to HIVE-27421

Reply:
Added a comment. Done.